lib/config/config.js

```diff
@@ -479,3 +479,25 @@ exports.getConfigList = function () {
 exports.setLogger = function (loggerInstance) {
 	logger = loggerInstance.context('config');
 };
+
+exports.expose = function (mageInstance) {
+	const httpServer = mageInstance.core.httpServer;
+
+	function handler(req, res, path, query) {
+		if (!query.appName) {
+			res.setHeader();
+			res.writeHead(400, {
+				'content-length': '0'
+			});
+			res.end();
+		}
+
+		const config = mageInstance.getConfig(query.appName);
+		res.writeHead(200, {
+			'content-type': 'application/json'
+		});
+		res.end(JSON.stringify(config));
```

Comment:
Why not cache the stringified value? Configuration will not change over a given runtime normally.